src/sqlitedb.cpp

```diff
@@ -225,13 +228,59 @@ bool DBBrowserDB::tryEncryptionSettings(const QString& filename, bool* encrypted
         {
             sqlite3_finalize(vm);
 #ifdef ENABLE_SQLCIPHER
+            if (!isDatabasePasswordChecked)
+            {
+                isDatabasePasswordChecked = true;
+
+                QMap<QString, QVariant> databasePasswords = PreferencesDialog::getSettingsValue("db", "databasepasswords").toMap();
```

Comment:
What about habing multiple files with the same name but in different directories? They might have the same password but they don't have to. So it might be necessary to allow the user to have different password for the same file name (i.e. having multiple entries with the same key). That means a QMap doesn't suffice anymore because there can be multiple values for the same key. A QMultiMap might be better so we can loop over all the values and try each until we find a working one.